samples/diagnostics/program-visualization/TraceExample.ipynb

```diff
@@ -0,0 +1,553 @@
+       "given operation using the QuantumSimulator.</p>\n",
+       "<h4>Required parameters</h4>\n",
+       "<ul>\n",
+       "<li>Q# operation or function name. This must be the first parameter, and must be a valid Q# operation\n",
+       "or function name that has been defined either in the notebook or in a Q# file in the same folder.</li>\n",
+       "<li>Arguments for the Q# operation or function must also be specified as <code>key=value</code> pairs.</li>\n",
+       "</ul>\n",
+       "<h4>Optional parameters</h4>\n",
+       "<ul>\n",
+       "<li><code>--depth=&lt;integer&gt;</code> (default=1): The depth at which to render operations along\n",
+       "the execution path.</li>\n",
+       "</ul>\n",
+       "<h5>Example</h5><p>Visualize the execution path of a Q# operation defined as <code>operation MyOperation() : Result</code>:</p>\n",
+       "<pre><code>In []: %trace MyOperation\n",
+       "Out[]: &lt;HTML visualization of the operation&gt;\n",
+       "</code></pre>\n",
+       "<h5>Example</h5><p>Visualize the execution path of a Q# operation defined as <code>operation MyOperation(a : Int, b : Int) : Result</code>:</p>\n",
+       "<pre><code>In []: %trace MyOperation a=5 b=10\n",
+       "Out[]: &lt;HTML visualization of the operation&gt;\n",
+       "</code></pre>\n",
+       "<h5>Example</h5><p>Visualize operations at depth 2 on the execution path of a Q# operation defined\n",
+       "as <code>operation MyOperation() : Result</code>:</p>\n",
+       "<pre><code>In []: %trace MyOperation --depth=2\n",
+       "Out[]: &lt;HTML visualization of the operation&gt;\n",
+       "</code></pre>\n"
+      ],
+      "text/plain": [
+       "%trace:\n",
+       "Visualizes the execution path of the given operation."
+      ]
+     },
+     "execution_count": 1,
+     "metadata": {},
+     "output_type": "execute_result"
+    }
+   ],
+   "source": [
+    "%trace?"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {
+    "slideshow": {
+     "slide_type": "slide"
+    }
+   },
+   "source": [
+    "### Basic operations\n",
+    "To start off, let's see how we can debug basic operations with the `%trace` magic command. First, let's define an operation `BasicOp` that will perform an `H` on the first qubit and an `X` on the second."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 2,
+   "metadata": {
+    "slideshow": {
+     "slide_type": "slide"
+    }
+   },
+   "outputs": [
+    {
+     "data": {
+      "application/json": "[\"BasicOp\"]",
+      "text/html": [
+       "<ul><li>BasicOp</li></ul>"
+      ],
+      "text/plain": [
+       "BasicOp"
+      ]
+     },
+     "execution_count": 2,
+     "metadata": {},
+     "output_type": "execute_result"
+    }
+   ],
+   "source": [
+    "operation BasicOp() : Unit {\n",
```

Comment:
I'd suggest renaming this operation to follow the [Q# style guide](https://docs.microsoft.com/quantum/resources/contributing/style-guide?tabs=guidance), so that we're showing a consistent view of the language to our users. In particular, "BasicOp" sounds like a noun (function) rather than a verb (operation).